src/deriving.ml

```diff
@@ -654,6 +671,12 @@ let wrap_sig ~loc ~hide sg =
   in
   if wrap then wrap_sig ~loc ~hide sg else sg
 
+let wrap_sig ~loc ~hide list =
+  List.concat_map list ~f:(fun { items; unused_code_warnings } ->
+    if List.is_empty items
+    then []
```

Comment:
This test is to avoid adding removing warnings in an empty structure, in case no code is generated, such as in:
```
include struct [@@@ocaml.warning "-32"] end[@@ocaml.doc "@inline"][@@merlin.hide]
```
That was not the original behaviour: the useless `inline struct` was added in any case. So this is an unrelated bonus feature from this PR!

However, those chains of shadowing can be hard to read: we now have three `wrap_sig` and `wrap_str` one after the other, and we don't know which function does what.
Maybe you could add at least a comment on what are the invariant the function expect, and what it does?
For instance, the last `wrap_sig` function is handling the empty case, and the middle `wrap_sig` invariant is that it takes a non-empty signature.